test/ariaqueryhandler.spec.ts

```diff
@@ -173,6 +174,305 @@ describeChromeOnly('AriaQueryHandler', () => {
+    });
+
+    it('hidden should wait for removal', async () => {
+      const { page } = getTestState();
+
+      await page.setContent(`<div role='main'></div>`);
+      let divRemoved = false;
+      const waitForSelector = page
+        .waitForSelector('aria/[role="main"]', { hidden: true })
+        .then(() => (divRemoved = true));
+      await page.waitForSelector('aria/[role="main"]'); // do a round trip
+      expect(divRemoved).toBe(false);
+      await page.evaluate(() => document.querySelector('div').remove());
+      expect(await waitForSelector).toBe(true);
+      expect(divRemoved).toBe(true);
+    });
+
+    it('should return null if waiting to hide non-existing element', async () => {
+      const { page } = getTestState();
+
+      const handle = await page.waitForSelector('aria/non-existing', {
+        hidden: true,
+      });
+      expect(handle).toBe(null);
+    });
+
+    it('should respect timeout', async () => {
+      const { page, puppeteer } = getTestState();
+
+      let error = null;
+      await page
+        .waitForSelector('aria/[role="button"]', { timeout: 10 })
+        .catch((error_) => (error = error_));
+      expect(error).toBeTruthy();
+      expect(error.message).toContain(
+        'waiting for selector `[role="button"]` failed: timeout'
+      );
+      expect(error).toBeInstanceOf(puppeteer.errors.TimeoutError);
+    });
+
+    it('should have an error message specifically for awaiting an element to be hidden', async () => {
+      const { page } = getTestState();
+
+      await page.setContent(`<div role='main'></div>`);
+      let error = null;
+      await page
+        .waitForSelector('aria/[role="main"]', { hidden: true, timeout: 10 })
+        .catch((error_) => (error = error_));
+      expect(error).toBeTruthy();
+      expect(error.message).toContain(
+        'waiting for selector `[role="main"]` to be hidden failed: timeout'
+      );
+    });
+
+    it('should respond to node attribute mutation', async () => {
+      const { page } = getTestState();
+
+      let divFound = false;
+      const waitForSelector = page
+        .waitForSelector('aria/zombo')
+        .then(() => (divFound = true));
+      await page.setContent(`<div aria-label='notZombo'></div>`);
+      expect(divFound).toBe(false);
+      await page.evaluate(() =>
+        document.querySelector('div').setAttribute('aria-label', 'zombo')
+      );
+      expect(await waitForSelector).toBe(true);
+    });
+
+    it('should return the element handle', async () => {
+      const { page } = getTestState();
+
+      const waitForSelector = page.waitForSelector('aria/zombo');
+      await page.setContent(`<div aria-label='zombo'>anything</div>`);
+      expect(
+        await page.evaluate(
+          (x: HTMLElement) => x.textContent,
+          await waitForSelector
+        )
+      ).toBe('anything');
+    });
+
+    // TODO: Figure out what the stack trace should look like for custom handlers
```

Comment:
What does it look like currently? Can you make a recommendation?

Reply:
This is the current stack trace:
```
"TimeoutError: waiting for selector `zombo` failed: timeout 10ms exceeded
    at new WaitTask (.../puppeteer/lib/cjs/puppeteer/common/DOMWorld.js:494:34)
    at DOMWorld.waitForSelectorInPage (.../puppeteer/lib/cjs/puppeteer/common/DOMWorld.js:427:26)
    at Object.waitFor (.../puppeteer/lib/cjs/puppeteer/common/AriaQueryHandler.js:55:21)
    at process._tickCallback (internal/process/next_tick.js:68:7)"
```
It would be nice if `ariaqueryhandler.spec.ts` was part of the trace since that is the actual call site for `waitForSelector`.
The fact that the selector is `zombo` instead of `aria/zombo` is maybe a bit unfortunate as well.

Reply:
It seems like we can't do much better than this since waitFor is async.